### DISCUSSION TITLE
Allow disabling "Dismiss stale pull request approvals when new commits are pushed" even when enabling "Whether the most recent reviewable push must be approved by someone other than the person who pushed it."

### DISCUSSION THREAD

**Discussion by jdimatteo** [community/community#77578] [Pull Requests]
### Select Topic Area

Bug

### Body

In my repo's "Protect matching branches" settings (at https://github.com/<my-org>/<my-repo>/settings/branch_protection_rules/<my-rule-number>) I have:

1. not checked / disabled: "Dismiss stale pull request approvals when new commits are pushed. New reviewable commits pushed to a matching branch will dismiss pull request review approvals."
2. checked / enabled: "Require approval of the most recent reviewable push. Whether the most recent reviewable push must be approved by someone other than the person who pushed it."

![image](https://github.com/community/community/assets/3752732/a7d92b19-6735-4cc2-838b-5c5ad0f35b55)

In my pr (https://github.com/<my-org>/<my-repo>/pull/<my-pr-number>) my coworker @darist approved the PR. At the bottom of the PR it showed "Changes approved" and the button "Squash and merge" was present:

![image](https://github.com/community/community/assets/3752732/03b0e7b8-8139-4319-8563-da0b4bc1cfcb)

Then I pushed a commit, and the bottom of the PR changed to "Review required. New changes require approval from someone other than the last pusher."

![image](https://github.com/community/community/assets/3752732/c0b2399c-d7c1-4312-88b5-0c6021ec63a9)

After unchecking (2), it changed back to "Changes approved" and I successfully merged the PR.
![image](https://github.com/community/community/assets/3752732/b3dc090f-739b-49a3-8774-8a4cf4f8002c)

How do I configure my repo to respect disabling (1) "Dismiss stale" even when (2) "must be approved by someone other than the person who pushed it" is checked?

These are two separate settings and arguably shouldn't be intertwined like this -- this is arguably a bug and GH behavior should change so that these settings do result in this PR remaining approved and mergable.

**Comment by jdimatteo** [comment]
Here is an example in a public repo:

1. I configured https://github.com/jdimatteo/test-repo-settings with these settings:
    1. not checked / disabled: "Dismiss stale pull request approvals when new commits are pushed. New reviewable commits pushed to a matching branch will dismiss pull request review approvals."
    2. checked / enabled: "Require approval of the most recent reviewable push. Whether the most recent reviewable push must be approved by someone other than the person who pushed it."
    
![image](https://github.com/community/community/assets/3752732/573177e1-438e-4790-a8e5-6eb1e12668f7)

    
2. Created https://github.com/jdimatteo/test-repo-settings/pull/26 with author jdimatteo@ with single commit https://github.com/jdimatteo/test-repo-settings/pull/26/commits/b8bae98c54b00b9c508b923155c890a81beccaa2

3. foo1111@ approved with a comment requesting a change to punctuation, showing "All checks have passed":

![image](https://github.com/community/community/assets/3752732/bdd8ec9e-d095-4bd9-b60f-8acae2c68a77)

4. jdimatteo@ addressed the punctuation review feedback pushing commit https://github.com/jdimatteo/test-repo-settings/pull/26/commits/6ccb8b2ce7986706aa7a2e41e415c0826a66169c

**expected behavior**: PR remains showing "All checks have passed"
**actual behavior**: PR shows "Review required" and "New changes require approval from someone other than the last pusher."

![image](https://github.com/community/community/assets/3752732/37d4d4b7-ea5f-4ca0-b950-965f9e786e4a)

This behavior is not correct because foo1111@ approved and jdimatteo@ is the last pusher, so the PR already has approval from someone other than the last pusher.

**Comment by ftruzzi** [comment]
@jdimatteo I completely agree with you. I think this feature should be named, as it initially was, "Require approval from someone other than the last pusher". The purpose would be to avoid someone hijacking your PR by pushing a bad commit and then approving it themselves.

Otherwise, as you said, there is a lot of overlap with the "Dismiss stale pull request approvals when new commits are pushed" feature and it's quite confusing.

I've opened a ticket to GitHub pointing this out and asking them to look at this discussion, but received no reply yet.

**Comment by jdimatteo** [comment]
Prior example was possibly wrong because approver was not in CODEOWNERS -- here is a correct example:

1. I configured https://github.com/jdimatteo/test-repo-settings with these settings:
    1. not checked / disabled: "Dismiss stale pull request approvals when new commits are pushed. New reviewable commits pushed to a matching branch will dismiss pull request review approvals."
    2. checked / enabled: "Require approval of the most recent reviewable push. Whether the most recent reviewable push must be approved by someone other than the person who pushed it."
    
![image](https://github.com/community/community/assets/3752732/573177e1-438e-4790-a8e5-6eb1e12668f7)

    
2. Created https://github.com/jdimatteo/test-repo-settings/pull/27 with author foo1111@ with single commit https://github.com/jdimatteo/test-repo-settings/pull/27/commits/630aedde301ac0b4091663708fa11c2c49351934

3. jdimatteo@ approved with a comment requesting a change to punctuation, showing "All checks have passed":

![image](https://github.com/community/community/assets/3752732/6d8da128-bd3a-492a-90e0-590b510796a4)

4. foo1111@ addressed the punctuation review feedback pushing commit https://github.com/jdimatteo/test-repo-settings/pull/27/commits/8d04e955b39f2b2fd9d226ec5d73648b6afbcbc0

**expected behavior**: PR remains showing "All checks have passed"
**actual behavior**: PR shows "Review required" and "New changes require approval from someone other than the last pusher."

![image](https://github.com/community/community/assets/3752732/40f4b504-c6a9-421f-859a-b17beea5a2ef)

This behavior is not correct because jdimatteo@ approved and foo1111@ is the last pusher, so the PR already has approval from someone other than the last pusher.

**Comment by jdimatteo** [comment]
FYI after discussing with GitHub enterprise support they said this is working as designed and they have no intention of changing the behavior nor clarifying the documentation / UI messaging. This is disappointing.
